dashboard/test/controllers/registrations_controller_test.rb

```diff
@@ -199,6 +199,30 @@ class RegistrationsControllerTest < ActionController::TestCase
     assert_equal EmailPreference::ACCOUNT_SIGN_UP, email_preference[:source]
   end
 
+  test "create new student in eu fails when missing required fields" do
+    eu_student_params = @default_params.update(data_transfer_agreement_accepted: true)
+
+    assert_does_not_create(User) do
+      post :create, params: {user: eu_student_params}
+    end
+  end
+
+  test "create new student in eu succeeds with required fields" do
+    eu_student_params = @default_params.merge(
+      {
+        data_transfer_agreement_accepted: true,
+        data_transfer_agreement_request_ip: "127.0.0.1",
+        data_transfer_agreement_source: User::DATA_TRANSFER_AGREEMENT_SOURCE_TYPES,
+        data_transfer_agreement_kind: "0",
```

Comment:
Comment somewhere about what this 0 means, please.  I'm still not sure if 0 is the right code for the dialog.